### PR TITLE
docs: Update Rust docs and snippets for upcoming release

### PR DESCRIPTION
**Pull request by Swatinem**
* Updates and fixes the code snippets for the upcoming Rust SDK release.
* Removes most of the sub-pages, and rather direct people to `docs.rs`.

### CODE REVIEW

File: src/collections/_documentation/error-reporting/capture-error/rust.md [hunk 1]
```diff
@@ -1,13 +1,24 @@
-In Rust you can capture errors that implement the `Fail` trait or that are `failure::Error`
-objects:
+In Rust you can capture any `std::error::Error` type.
```
Comment (MimiDumpling) [inline src/collections/_documentation/error-reporting/capture-error/rust.md:1]:
```suggestion
In Rust, you can capture any `std::error::Error` type. 
```

**Comment by MimiDumpling** [comment]
LGTM 👍 I'll leave approval to @jan-auer for the more technical aspects of the doc

File: src/collections/_documentation/error-reporting/configuration/drain-example/rust.md [hunk 2]
```diff
@@ -4,9 +4,8 @@ the guard and make sure it disposes on shutdown.  Alternatively the client can b
 
 ```rust
 use std::time::Duration;
-use sentry::Hub;
 
-if let Some(client) = Hub.current().client() {
+if let Some(client) = sentry::Hub::current().client() {
```
Comment (jan-auer) [inline src/collections/_documentation/error-reporting/configuration/drain-example/rust.md:8]: _nit_: I'd slightly prefer to keep the import.

File: src/collections/_documentation/platforms/rust/index.md [hunk 3]
```diff
@@ -10,36 +10,26 @@ events to Sentry.
...
-* [failure]({% link _documentation/platforms/rust/failure.md %})
-* [error-chain]({% link _documentation/platforms/rust/error_chain.md %})
-
-Additionally you can catch panics using the panic integration:
-
-* [panic]({% link _documentation/platforms/rust/panic.md %})
-
-### Logging integrations
+A list of integrations and their feature flags can be found
+[here in the API docs](https://docs.rs/sentry/0/sentry/integrations/index.html).
```
Comment (jan-auer) [inline src/collections/_documentation/platforms/rust/index.md:28]:
```suggestion
[in the integration API docs](https://docs.rs/sentry/0/sentry/integrations/index.html).
```

File: src/collections/_documentation/platforms/rust/index.md [hunk 4]
```diff
@@ -10,36 +10,26 @@ events to Sentry.
...
 
-Logs can be automatically converted into breadcrumbs.
+Apart from those, Sentry-Rust also supports the following application frameworks:
```
Comment (jan-auer) [inline src/collections/_documentation/platforms/rust/index.md:30]:
```suggestion
Apart from those, the Rust SDK also supports the following application frameworks:
```

**Pull request merged**